### ISSUE TITLE
When a job fails submission it dies painfully

### ISSUE THREAD

**Issue by jerowe** [jerowe/HPC-Runner-Command#9]
```
[2017/05/31 06:07:41] Submited job /scratch/gencore/Youssef_data/hpc-runner/part2_gatk/scratch/part2_gatk_023_bcftools_stats.sh
        With Slurm jobid 269097 
[2017/05/31 06:07:42] sbatch: error: Batch job submission failed: Job violates accounting/QOS policy (job submit limit, user's size and/or time limits)
 
[2017/05/31 06:07:47] Job was not submitted successfully 
[2017/05/31 06:07:47] STDERR: sbatch: error: Batch job submission failed: Job violates accounting/QOS policy (job submit limit, user's size and/or time limits)
 
[2017/05/31 06:07:47] Submit scripts will be written, but will not be submitted to the queue. 
[2017/05/31 06:07:47] Please look at your submission scripts in /scratch/gencore/Youssef_data/hpc-runner/part2_gatk/scratch 
[2017/05/31 06:07:47] And your logs in /scratch/gencore/Youssef_data/hpc-runner/part2_gatk/logs/2017-05-31-hpcrunner_logs
for more information 
[2017/05/31 06:07:47] There are 500 tasks for job type bcftools_stats 
[2017/05/31 06:07:47] Submitted in 3 job arrays.
 
Attribute (scheduler_id) does not pass the type constraint because: Validation failed for 'Str' with value undef at accessor HPC::Runner::Command::submit_jobs::Utils::Scheduler::Batch::scheduler_id (defined at /scratch/gencore/.local/easybuild/software/gencore_dev/1.0/lib/perl5/site_perl/5.22.0/HPC/Runner/Command/submit_jobs/Utils/Scheduler/Batch.pm line 63) line 4
```

**Comment by jerowe** [comment]
Additionally, when it does this the task dependencies are never created because it DIES.

**Comment by jerowe** [comment]
This is fixed in 3.2.3.

When a job fails it gives an error message. Additionally, any other jobs that depend upon that job will also not be submitted.
